controllers/novels.js

```diff
@@ -0,0 +1,90 @@
+const models = require('../models')
+
+const getAllNovelsWithAuthorAndGenre = async (request, response) => {
+  try {
+    const allNovels = await models.Novels.findAll({
+      include: [
+        { model: models.Authors },
+        { model: models.Genres }]
+    })
+
+    allNovels.forEach(novel => novel.genres.sort((lhs, rhs) => {
+      if (lhs.name < rhs.name) {
+        return -1
+      }
+      else if (lhs.name > rhs.name) {
+        return 1
+      }
+      else {
+        return 0
+      }
+    }))
+
+    return (allNovels)
+      ? response.send(allNovels)
+      : response.sendStatus(404)
+  } catch (error) {
+    response.status(500).send('Unable to retrieve novels, please try again')
+  }
+}
+
+const getNovelbySlugType = (request, response) => {
+  const { slug } = request.params
```

Comment:
This isn't actually a slug, a slug is a specific type of value. This is either an id or a last name so something like `identifier` or `searchTerm` is a better name